driver/format/RowBinaryWithNamesAndTypes.cpp

```diff
@@ -28,41 +28,34 @@ RowBinaryWithNamesAndTypesResultSet::RowBinaryWithNamesAndTypesResultSet(std::is
             columns_info[i].type_without_parameters = "String";
         }
 
-        columns_info[i].updateTypeId();
-
-        if (columns_info[i].display_size == 0) {
-            auto type_name = convertTypeIdToUnparametrizedCanonicalTypeName(columns_info[i].type_without_parameters_id);
-
-            if (
-                type_name == "Decimal32" &&
-                type_name == "Decimal64" &&
-                type_name == "Decimal128"
-            ) {
-                type_name = "Decimal";
-            }
-
-            if (type_name == "FixedString") {
-                columns_info[i].display_size = columns_info[i].fixed_size;
-            }
-            else if (type_name == "String") {
-                columns_info[i].display_size = 0; // Will be populated when the actual data is received.
-            }
-            else {
-                auto & type_info = type_info_for(type_name);
-                columns_info[i].display_size = type_info.column_size;
-            }
-        }
+        columns_info[i].updateTypeInfo();
     }
 
     finished = columns_info.empty();
-
-    constexpr std::size_t prefetch_at_least = 10'000;
-    tryPrefetchRows(prefetch_at_least);
 }
 
 bool RowBinaryWithNamesAndTypesResultSet::readNextRow(Row & row) {
-    if (raw_stream.peek() == EOF)
+    if (raw_stream.peek() == EOF) {
```

Comment:
Looks like excact copy of 'ODBCDriver2ResultSet::readNextRow', copy-paste ? Could you factor out the `column_info.display_size` recomputation to a fucntion ?

Reply:
Done. Moved it to one common place. Thanks.